---
# Import libraries Requried
---

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
# Import other models as needed


---
# Loading Data and Spliting Data

 * to Test
 * to Train
 ---

In [2]:
# Load preprocessed data
file_path = '/content/drive/My Drive/Airline_Reviews_Preprocessed.csv'
df = pd.read_csv(file_path)

# Split data into features (X) and target (y)
X = df['cleaned_reviews']
y = df['label']

# Split into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


---
# Initialize Models

testing different models and Obtaining Test Results.

Models that are Tested:
 * Logistic Regression
 * Support Vector Machine
 * Random Forest
 ---

In [3]:
# Initialize models
models = {
    'Logistic Regression': LogisticRegression(),
    'Support Vector Machine': SVC(),
    'Random Forest': RandomForestClassifier()
    # Add more models as needed
}

# Train and evaluate each model
results = {}
for model_name, model in models.items():
    # Vectorize text data
    vectorizer = TfidfVectorizer(max_features=5000)
    X_train_vec = vectorizer.fit_transform(X_train).toarray()
    X_test_vec = vectorizer.transform(X_test).toarray()

    # Train the model
    model.fit(X_train_vec, y_train)

    # Predictions
    y_pred = model.predict(X_test_vec)

    # Evaluate
    accuracy = accuracy_score(y_test, y_pred)
    precision = precision_score(y_test, y_pred)
    recall = recall_score(y_test, y_pred)
    f1 = f1_score(y_test, y_pred)

    results[model_name] = {
        'Accuracy': accuracy,
        'Precision': precision,
        'Recall': recall,
        'F1 Score': f1
    }

    # Print results
    print(f"Results for {model_name}:")
    print(f"Accuracy: {accuracy:.4f}, Precision: {precision:.4f}, Recall: {recall:.4f}, F1 Score: {f1:.4f}")
    print()

# Optionally, save results or further analyze them


Results for Logistic Regression:
Accuracy: 0.9066, Precision: 0.9143, Recall: 0.9475, F1 Score: 0.9306

Results for Support Vector Machine:
Accuracy: 0.9124, Precision: 0.9238, Recall: 0.9455, F1 Score: 0.9345

Results for Random Forest:
Accuracy: 0.8872, Precision: 0.8944, Recall: 0.9403, F1 Score: 0.9168



---
# Results:

Results for Logistic Regression:
* Accuracy: 0.9066
* Precision: 0.9143
* Recall: 0.9475
* F1 Score: 0.9306

Results for Support Vector Machine:
* Accuracy: 0.9124
* Precision: 0.9238
* Recall: 0.9455
* F1 Score: 0.9345

Results for Random Forest:
* Accuracy: 0.8872
* Precision: 0.8944
* Recall: 0.9403
* F1 Score: 0.9168
---

*** THE SUPPORT VECTOR MACHINE PERFORMED WELL ***

---